transports/tcp/src/lib.rs

```diff
@@ -128,15 +124,26 @@ impl TcpConfig {
     }
 }
 
-impl Transport for TcpConfig {
+#[derive(Clone, Debug, Default)]
+pub struct TcpListener {
```

Comment:
Why not implement `Listener` and `Dialer` directly on `TcpConfig`?

Reply:
I moved away from a single type for both traits in this PR to have a more natural API for `core::Transport`, e.g. I considered it better to write:


```rust
let _t = Transport::denied()
    .with_dialer(DnsDialer::new(TcpDialer::default())
    .with_listener(WsListener::new(TcpListener::default()));
```

rather than:

```rust
let _t = Transport::denied()
    .with_dialer(DnsConfig::new(TcpConfig::default())
    .with_listener(WsConfig::new(TcpConfig::default()));
```

I.e. the single `*Config` type emphasises the combined listener and dialer which I thought of as coming together only in `core::Transport` (or some user-defined type which is built out of dialers and listeners, or just dialers or only listeners). Seeing a `*Dialer` or `*Listener` in isolation immediately conveyed their respective role, whereas a `*Config` could serve multiple masters.